operators/src/plot/statistics.rs

```diff
@@ -41,79 +49,228 @@ impl PlotOperator for Statistics {
+    source: Op,
 }
 
-impl InitializedPlotOperator for InitializedStatistics {
+impl<Op> InitializedStatistics<Op> {
+    pub fn new(
+        result_descriptor: PlotResultDescriptor,
+        column_names: Vec<String>,
+        source: Op,
+    ) -> Self {
+        Self {
+            result_descriptor,
+            column_names,
+            source,
+        }
+    }
+}
+
+impl InitializedPlotOperator for InitializedStatistics<Box<dyn InitializedVectorOperator>> {
+    fn result_descriptor(&self) -> &PlotResultDescriptor {
+        &self.result_descriptor
+    }
+
+    fn query_processor(&self) -> Result<TypedPlotQueryProcessor> {
+        Ok(TypedPlotQueryProcessor::JsonPlain(
+            StatisticsVectorQueryProcessor {
+                vector: self.source.query_processor()?,
+                column_names: self.column_names.clone(),
+            }
+            .boxed(),
+        ))
+    }
+}
+
+impl InitializedPlotOperator for InitializedStatistics<Vec<Box<dyn InitializedRasterOperator>>> {
+    fn result_descriptor(&self) -> &PlotResultDescriptor {
+        &self.result_descriptor
+    }
+
     fn query_processor(&self) -> Result<TypedPlotQueryProcessor> {
         Ok(TypedPlotQueryProcessor::JsonPlain(
-            StatisticsQueryProcessor {
+            StatisticsRasterQueryProcessor {
                 rasters: self
-                    .rasters
+                    .source
                     .iter()
                     .map(InitializedRasterOperator::query_processor)
                     .collect::<Result<Vec<_>>>()?,
+                column_names: self.column_names.clone(),
             }
             .boxed(),
         ))
     }
+}
 
-    fn result_descriptor(&self) -> &PlotResultDescriptor {
-        &self.result_descriptor
+/// A query processor that calculates the statistics about its vector input.
+pub struct StatisticsVectorQueryProcessor {
+    vector: TypedVectorQueryProcessor,
+    column_names: Vec<String>,
+}
+
+#[async_trait]
+impl PlotQueryProcessor for StatisticsVectorQueryProcessor {
+    type OutputFormat = serde_json::Value;
+
+    fn plot_type(&self) -> &'static str {
+        STATISTICS_OPERATOR_NAME
+    }
+
+    async fn plot_query<'a>(
+        &'a self,
+        query: PlotQueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> Result<Self::OutputFormat> {
+        let mut number_statistics: HashMap<String, NumberStatistics> = self
+            .column_names
+            .iter()
+            .map(|column| (column.clone(), NumberStatistics::default()))
+            .collect();
+
+        call_on_generic_vector_processor!(&self.vector, processor => {
+            let mut query = processor.query(query, ctx).await?;
+
+            while let Some(collection) = query.next().await {
+                let collection = collection?;
+
+                for (column, stats) in &mut number_statistics {
+                    collection.data(column).expect("checked in param").float_options_iter().for_each(
```

Comment:
Maybe we can code defensive here and say:
If it does not exist, just `continue` (instead of panic).

Reply:
Can this even happen after checking that the column exists in line 127?

Reply:
Well, it is in the result descriptor and thus in 99,999999% of the cases should be part of the feature collection (if there is no bug, etc.). But if it wouldn't be there for whatever reasons, it is defensive to code it to not panic.

Sometimes it is hard to do so, but here it should be very easy.

Reply:
NaN is now assumed for all values in an unexpectedly missing column (https://github.com/geo-engine/geoengine/pull/614/commits/b12212ea766ec55d63406194a8b1109136421add).
Let me know if you prefer to completely exclude the missing column values from the statistics (total count).

Also note that some other plot operators (scatter plot, box plot, ...) panic in this case. It should probably also be changed there.